src/algorithms/configuration/maps/dense_map.py

```diff
@@ -24,27 +24,26 @@ class DenseMap(Map):
     """
     grid: np.array
 
-    #The transpose flag is set to true as a default for the initialization, as that is 
-    #how we are storing internally, but it will be set to false when we are simply translating from sparsemap
-    #When we create more map views, we should set it to false
-    def __init__(self, grid: Optional[List], services: Services = None, transpose: bool = True) -> None:
-        self.grid = None 
+    # The transpose flag is set to true as a default for the initialization, as that is
+    # how we are storing internally, but it will be set to false when we are simply translating from sparsemap
+    # When we create more map views, we should set it to false
+    def __init__(self, grid: Optional[List] = None, services: Services = None, transpose: bool = True, mutable: bool = False) -> None:
+        self.grid = None
 
         arr_grid = None
         if grid is not None:
             arr_grid = np.atleast_2d(np.array(grid))
-            super().__init__(Size(*([0]*arr_grid.ndim)), services)
+            super().__init__(Size(*([0]*arr_grid.ndim)), services, mutable)
         else:
-            super().__init__(Size(0, 0), services)
+            super().__init__(services=services, mutable=mutable)
```

Comment:
Why did you get rid of `Size(0,0)` here?

Reply:
Because that's a retarded default.

Reply:
We had to initialise super as 2D, but since we're using `None` it does seem a little pointless now